llvm/test/Transforms/InstCombine/AMDGPU/addrspacecast.ll

```diff
@@ -0,0 +1,25 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -mtriple=amdgcn-amd-amdhsa -passes=instcombine %s | FileCheck %s
+
+; Check that `select B, true, C` isn't optimized to `or B, C`,
+; because the invalid addrspacecast %asc.shared may introduce poison.
```

Comment:
Description is slightly inaccurate. In this case it will always be poison

Reply:
> Description is slightly inaccurate. In this case it will always be poison

done